src/xtgeo/grid3d/_grdecl_format.py

```diff
@@ -134,13 +143,13 @@ def interpret_token(val):
 
 @contextmanager
 def open_grdecl(
-    grdecl_file,
-    keywords,
-    simple_keywords=[],
-    max_len=None,
-    ignore=IGNORE_ALL,
-    strict=True,
-):
+    grdecl_file: FileLike,
+    keywords: list[str],
+    simple_keywords: list[str] | None = None,
```

Comment:
Having mutable defaults in functions is not great, these become globals, and will easily introduce unexpected behavior.